spikeinterface/extractors/cellexplorersortingextractor.py

```diff
@@ -0,0 +1,136 @@
+            sorting_id = spikes_matfile_path.name.split(".")[0]
+            if session_info_matfile_path is None:
+                session_info_matfile_path = folder_path / f"{sorting_id}.sessionInfo.mat"
+            session_info_matfile_path = Path(session_info_matfile_path)
+            assert (
+                (session_info_matfile_path).is_file()
+            ), f"No {sorting_id}.sessionInfo.mat file found in the folder!" 
+
+            try:
+                session_info_mat = scipy.io.loadmat(file_name=str(session_info_matfile_path))
+                self.read_session_info_with_scipy = True
+            except NotImplementedError:
+                session_info_mat = hdf5storage.loadmat(file_name=str(session_info_matfile_path))
+                self.read_session_info_with_scipy = False
+            
+            assert session_info_mat["sessionInfo"]["rates"][0][0]["wideband"], (
+                "The sesssionInfo.mat file must contain "
+                "a 'sessionInfo' struct with field 'rates' containing field 'wideband' to extract the sampling frequency!"
+            )
+            if self.read_session_info_with_scipy:
+                sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+            else:
+                sampling_frequency = float(
+                    session_info_mat["sessionInfo"]["rates"][0][0]["wideband"][0][0]
+                )  # careful not to confuse it with the lfpsamplingrate; reported in units Hz
+
+        try:
+            spikes_mat = scipy.io.loadmat(file_name=str(spikes_matfile_path))
+            self.read_spikes_info_with_scipy = True
+        except NotImplementedError: 
+            spikes_mat = hdf5storage.loadmat(file_name=str(spikes_matfile_path))
+            self.read_spikes_info_with_scipy = False
+
+        assert np.all(
+            np.isin(["UID", "times"], spikes_mat["spikes"].dtype.names)
+        ), "The spikes.cellinfo.mat file must contain a 'spikes' struct with fields 'UID' and 'times'!"
+
+        # CellExplorer reports spike times in units seconds; SpikeExtractors uses time units of sampling frames
+        # Rounding is necessary to prevent data loss from int-casting floating point errors
+        if self.read_spikes_info_with_scipy:
+            unit_ids = np.asarray(spikes_mat["spikes"]["UID"][0][0][0])
+            spiketrains = [
+                (np.array([y[0] for y in x]) * sampling_frequency).round().astype(np.int64)
+                for x in spikes_mat["spikes"]["times"][0][0][0]
+            ]
+        else:
+            unit_ids = np.asarray(spikes_mat["spikes"]["UID"][0][0])
+            spiketrains = [
+                (np.array([y[0] for y in x]) * sampling_frequency).round().astype(np.int64)
+                for x in spikes_mat["spikes"]["times"][0][0]            
+            ]
```

Comment:
why not using a dict here to avoid the 
`spike_frames = self._spiketrains[self._unit_ids.index(unit_id)]`